caas/kubernetes/clientconfig/k8s.go

```diff
@@ -15,15 +16,18 @@ import (
 
 var logger = loggo.GetLogger("juju.caas.kubernetes.clientconfig")
 
-// K8SClientConfig parses Kubernetes client configuration from the default location or $KUBECONFIG.
-func K8SClientConfig() (*ClientConfig, error) {
+// NewClientConfig parses Kubernetes client configuration from the default location or $KUBECONFIG.
+func NewClientConfig(reader io.Reader) (*ClientConfig, error) {
 
-	configPath := getKubeConfigPath()
-
-	config, err := clientcmd.LoadFromFile(configPath)
+	content, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	config, err := parseKubeConfig(content)
 	if err != nil {
-		return nil, errors.Annotatef(err, "failed to read kubernetes config from '%s'", configPath)
+		return nil, errors.Trace(err)
```

Comment:
I think using errors.Annotate here would be good, eg
errors.Annotate(err, "error parsing Kubernetes config")